### ISSUE TITLE
How to prepare our own datasets for Segmentation ?

### ISSUE THREAD

**Issue by jay-thakur** [ranahanocka/MeshCNN#54]
Hi,

Thank you so much for this repo. This is really helpful & very impressive work.

I am able to run the code & produce the output as expected. However I am looking to run this code on shapenet dataset (having chairs, table etc...). Could you please help me to preprocess(prepare) our own dataset for segmentation. 

Also it would be great if you can share any such datasets link for mesh segmentation.

Thanks,
Jay

**Comment by ranahanocka** [comment]
Hi @jay-thakur ,

I think you said you were looking for how to create the .edges file that the segmentation dataset prep code expects right? You should export the .edges file from the Mesh class in the python code. Here is the deprecated function that did it:

```
def save_new_edges(filename):
    mesh = Mesh(filename, shuffle=False)
    edges = np.array(mesh.edges)
    edge_file = filename.replace('.obj', '.edges')
    np.savetxt(edge_file, edges, fmt='%d')
```

**Comment by Pavelevivan** [comment]
Hi,
Can you explain what is the order to get dataset for segmentation:
obj -> edges -> seg -> eseg (is that correct?)
I downloaded your scripts on matlab, but quite frustrated on them because I do not have experience with it. Could you explain me that? I would really appreciate your help.

Thanks,
Ivan

**Comment by yuvaramsingh94** [comment]
Hi,
i am also on the same page . i like to create the .seseg file for my own dataset . is there any python code available for it . as i am not that used to matlab. 

Also i like to know is there any code/strategy available to map .eseg annotation of one resolution .obj file to a lower resolution . i am planning to figure out the best resolution for my work so planning to label(.eseg) in higher resolution and convert it to other resolutions so i wont have to do the labeling all over again .

**Comment by ranahanocka** [comment]
> obj -> edges -> seg -> eseg (is that correct?)

Yes, that is correct. I think if you try to reproduce the example from the MATLAB scripts it will be clear

**Comment by ranahanocka** [comment]
Hi @yuvaramsingh94 ,

The matlab code maps the segmentation from one resolution to another (it computes some graph cut optimization to do that). Unfortunately, there is only matlab code to do this.

**Comment by KevinJianLin** [comment]
@jay-thakur , are you able to prepare segmentation file for your own datasets? I can't find Mesh class in the repo. Where is it under?

def save_new_edges(filename):
    mesh = Mesh(filename, shuffle=False)
    edges = np.array(mesh.edges)
    edge_file = filename.replace('.obj', '.edges')
    np.savetxt(edge_file, edges, fmt='%d')

**Comment by KevinJianLin** [comment]
In this function, filename is .obj file. I got this error, AttributeError: 'NoneType' object has no attribute 'num_aug'. 
Do I need to place my .obj file under specific folder to avoid this error?


def save_new_edges(filename):
    mesh = Mesh(filename, shuffle=False)
    edges = np.array(mesh.edges)
    edge_file = filename.replace('.obj', '.edges')
    np.savetxt(edge_file, edges, fmt='%d')

**Comment by KevinJianLin** [comment]
@jay-thakur does the function work for you? @ranahanocka  Is there previous commit to checkout save_new_edges function?

**Comment by jay-thakur** [comment]
Hi,

I had tried this 3-4 yrs ago, tbh I don't remember now.

Thanks,
Jay

**Comment by lssuking** [comment]
Hi @KevinJianLin ,
i got the same  AttributeError as you said above.And i checked the code and then found maybe some conflicts in the code.
After that did you have any ideals to deal with the error?
